internal/osbuild2/grub2_legacy_stage.go

```diff
@@ -0,0 +1,227 @@
+package osbuild2
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+
+	"github.com/osbuild/osbuild-composer/internal/common"
+	"github.com/osbuild/osbuild-composer/internal/disk"
+)
+
+type GRUB2FSDesc struct {
+	Device string     `json:"device,omitempty"`
+	Label  string     `json:"label,omitempty"`
+	UUID   *uuid.UUID `json:"uuid,omitempty"`
+}
+
+func (d GRUB2FSDesc) validate() error {
+
+	have := make([]string, 0, 3)
+	if d.Device != "" {
+		have = append(have, "`device`")
+	}
+
+	if d.Label != "" {
+		have = append(have, "`label`")
+	}
+
+	if d.UUID != nil {
+		have = append(have, "`uuid`")
+	}
+
+	count := len(have)
+	if count == 0 {
+		return fmt.Errorf("need `device`, `label` or `uuid`")
```

Comment:
Please add a comma after `label` ;)

Reply:
> Does RHEL 7 need a RHEL 8 build root for qemu? I seem to recall this was a requirement but I don't see it anywhere.

Yeah, I had the same though (also because I made a not w.r.t to that in the corresponding jira ticket). Will double check.

Reply:
Just checked: we have `qemu-img version 1.5.3`, it can do `subformat=fixed` but not `force_size`, which we also use for `vpc`.

```
[gicmo@ckellner-rhel7 ~]$ qemu-img convert -O vpc -o subformat=fixed,force_size disk.qcow2 disk.vpc
Unknown option 'force_size'
qemu-img: Invalid options for file format 'vpc'.
```

Reply:
I think we might need the `force_size` option, see https://repo.or.cz/qemu/kevin.git/blobdiff/798609bbe29df4edf4bbec46927fc12862b02378..fb9245c2610932d33ce148b58714fcc7b3c6eb5f:/block/vpc.c

Reply:
For future reference. Old `qemu-img` does not need the `force_size` option, so we made that optional in osbuild.